src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoAction.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.Util;
+import hudson.model.InvisibleAction;
+import java.io.ObjectStreamException;
+import java.net.URL;
+import jenkins.branch.MultiBranchProject;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.github.GHRepository;
+
+/**
+ * Invisible property on {@link MultiBranchProject}
+ * that retains information about GitHub repository.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class GitHubRepoAction extends InvisibleAction {
+    private final URL url;
+    private final String description;
+    private final String homepage;
+
+    public GitHubRepoAction(GHRepository repo) {
+        this(repo.getHtmlUrl(), repo.getDescription(), repo.getHomepage());
+    }
+
+    public GitHubRepoAction(URL url, String description, String homepage) {
+        this.url = url;
+        this.description = Util.fixEmpty(description);
+        this.homepage = Util.fixEmpty(homepage);
+    }
+
+    public GitHubRepoAction(GitHubRepoAction that) {
+        this(that.getUrl(), that.getDescription(), that.getHomepage());
+    }
+
+    private Object readResolve() throws ObjectStreamException {
+        if ((description != null && StringUtils.isBlank(description))
```

Comment:
Basically, during data migration, there are some cases where a deserialized object will have an empty description where the description should be null... this readResolve will fix those such that the description is then correctly null for those cases